### PR TITLE
docs: Adds example of Prisma generated type

### PR DESCRIPTION
**Pull request by guerra08**
This PR adds an example of **Prisma Client generated types** when using a **LoaderFunction**. This is similar to other samples found on the example projects and video guides on Remix YouTube channel.

image

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
You actually don't need these extra types

https://twitter.com/millsp_/status/1485931866586857479

**Comment by guerra08** [comment]
> You actually don't need these extra types
> 
> https://twitter.com/millsp_/status/1485931866586857479

Oh, nice! I have update the PR with this suggestion.

**Comment by ryanflorence** [comment]
@kentcdodds please review and merge when you're ready :)

File: docs/guides/data-loading.md [hunk 1]
```diff
@@ -161,6 +161,30 @@ export default function ProductCategory() {
...
+
+```tsx filename=tsx filename=app/routes/products/$productId.tsx
+import { useLoaderData } from "remix";
+import { db } from "~/db.server";
+
+type LoaderData = Awaited<ReturnType<typeof loader>>
+
+export let loader = async ({ params }) => {
+  return db.product.findMany({});
+};
```
Comment (kentcdodds) [inline docs/guides/data-loading.md:174]: The challenge here is that if you want to set headers you can no longer use the return value as it's a Response object instead. So how about this:

```suggestion
import { useLoaderData, json } from "remix";
import { db } from "~/db.server";

type LoaderData = Awaited<ReturnType<typeof getLoaderData>>

async function getLoaderData() {
  const products = await db.product.findMany({
    select: {
      id: true,
      name: true,
      imgSrc: true,
    }
  })
  return { products }
}

export let loader = async () => {
  return json<LoaderData>(await getLoaderData());
};
```

**Review by kentcdodds — APPROVED** [review]
Thanks for this! I'm going to make a small change here, but it's great 👍

**Comment by guerra08** [comment]
Thanks guys, happy to contribute!

**Pull request merged**